connectors/cassandra/schema.go

```diff
@@ -37,17 +55,197 @@ func (c *Connector) TruncateScope(ctx context.Context, scope string) error {
 
 // DropScope is not implemented
 func (c *Connector) DropScope(ctx context.Context, scope string) error {
-	panic("next diff")
+	ksn := CleanupKeyspaceName(scope)
+	err := c.Session.Query(fmt.Sprintf(`DROP KEYSPACE IF EXISTS "%s"`, ksn)).Exec()
+	if err != nil {
+		return errors.Wrapf(err, "Unable to drop keyspace %q", ksn)
+	}
+	return nil
+}
+
+// RepairableSchemaMismatchError is an error describing what can be added to make
+// this schema current. It might include a lot of tables or columns
+type RepairableSchemaMismatchError struct {
+	MissingColumns []MissingColumn
+	MissingTables  []string
+}
+
+// MissingColumn describes a column that is missing
+type MissingColumn struct {
+	Column    dosa.ColumnDefinition
+	Tablename string
 }
 
-// CheckSchema is not implemented
+// HasMissing returns true if there are missing columns
+func (m *RepairableSchemaMismatchError) HasMissing() bool {
+	return m.MissingColumns != nil || m.MissingTables != nil
+}
+
+// Error prints a human-readable error message describing the first missing table or column
+func (m *RepairableSchemaMismatchError) Error() string {
+	if m.MissingTables != nil {
+		return fmt.Sprintf("Missing %d tables (first is %q)", len(m.MissingTables), m.MissingTables[0])
+	}
+	return fmt.Sprintf("Missing %d columns (first is %q in table %q)", len(m.MissingColumns), m.MissingColumns[0].Column.Name, m.MissingColumns[0].Tablename)
+}
+
+// compareStructToSchema compares a dosa EntityDefinition to the gocql TableMetadata
+// There are two main cases, one that we can fix by adding some columns and all the other mismatches that we can't fix
+func compareStructToSchema(ed *dosa.EntityDefinition, md *gocql.TableMetadata, schemaErrors *RepairableSchemaMismatchError) error {
```

Comment:
You're essentially doing 3 checks in this function. First you check the partitions keys for validity. Then you check the clustering columns for validity, then you ensure all the columns exist. So what if this function was the following:
```
func compareStructToSchema(...) {
  if err = checkPartitionKeys(...); err != nil {
     return err
  } else if err = checkClusteringColumn(...); err != nil {
     return err
  } else if err = checkColumns(...); err != nil {
     return err
  }
  return nil
}
```
Then your high level algorithm reads really cleanly and the details of each check are encoded in smaller, more readable functions.